msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/MultipleAccountPublicClientTests.java

```diff
@@ -0,0 +1,539 @@
+            @Override
+            public void onTaskCompleted(List<IAccount> result) {
+                accountsUnderTest.addAll(result);
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                fail(exception.getMessage());
+            }
+        });
+
+        RoboTestUtils.flushScheduler();
+
+        // act
+        final ResultFuture<IAuthenticationResult> future = new ResultFuture<>();
+        try {
+            mMultipleAccountPCA.acquireTokenSilentAsync
+                    (getScopes(),
+                            accountsUnderTest.get(0),
+                            testCaseData.userAccountsData.get(0).cloud +
+                                    "/" + testCaseData.userAccountsData.get(0).tenantId,
+                            new SilentAuthenticationCallback() {
+                                @Override
+                                public void onSuccess(IAuthenticationResult authenticationResult) {
+                                    future.setResult(authenticationResult);
+                                }
+
+                                @Override
+                                public void onError(MsalException exception) {
+                                    future.setException(exception);
+                                }
+                            });
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+
+        RoboTestUtils.flushScheduler();
+
+        // assert
+        IAuthenticationResult authenticationResult = null;
+        try {
+            authenticationResult = future.get();
+            assertNotNull("authenticationResult from acquireTokeSilent call", authenticationResult);
+            assertEquals("accessToken value",
+                    testCaseData.userAccountsData.get(0).getFakeAccessToken(), authenticationResult.getAccessToken());
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }
+
+    private void performInteractiveAcquireTokenWithMockResponsesToSetupCache() {
+        TestUtils.clearCache(SHARED_PREFERENCES_NAME);
+        MockHttpClient.reset();
+        MockHttpClient.setHttpResponse(
+                MockServerResponse.getMockCloudDiscoveryResponse(),
+                HttpClient.HttpMethod.GET,
+                MockTokenCreator.CLOUD_DISCOVERY_ENDPOINT_REGEX);
+        for (UserAccountData userData: testCaseData.userAccountsData) {
+            HttpResponse mockResponseForHomeTenant = MockServerResponse.getMockTokenSuccessResponse(
+                    userData.localAccountId,
+                    userData.tenantId,
+                    userData.cloud,
+                    testCaseData.getRawClientInfo(),
+                    userData.getFakeAccessToken());
+            MockHttpClient.setHttpResponse(mockResponseForHomeTenant,
+                    HttpClient.HttpMethod.POST,
+                    userData.cloud + "/.*");
```

Comment:
Might be better to have this entire code in its own method such as `setupMockedHttpClientResponses` and then you can this and `performInteractiveAcquireTokenCall` in the `before` as opposed to calling this one method which seems to be doing two separate things into one method.....might be better to separate...but your call

Reply:
done